lib/ansible/module_utils/oneview.py

```diff
@@ -132,6 +128,69 @@ def _standardize_value(value):
     return str(value)
 
 
+class OneViewModuleException(Exception):
+    """
+    OneView base Exception.
+
+    Attributes:
+       msg (str): Exception message.
+       oneview_response (dict): OneView rest response.
+   """
+
+    def __init__(self, data):
+        self.msg = None
+        self.oneview_response = None
+
+        if isinstance(data, six.string_types):
+            self.msg = data
+        else:
+            self.oneview_response = data
+
+            if data and isinstance(data, dict):
+                self.msg = data.get('message')
+
+        if self.oneview_response:
+            Exception.__init__(self, self.msg, self.oneview_response)
+        else:
+            Exception.__init__(self, self.msg)
```

Comment:
To be honest, I would leave it up to the Exception object to check if the 3 parameter is fine, rather than make 2 almost identical calls here. But that's just a mental note I made, not a requirement per se.